src/biokbase/narrative/jobs/jobcomm.py

```diff
@@ -181,14 +187,21 @@ def __init__(self):
                 MESSAGE_TYPE["STOP_UPDATE"]: self._modify_job_updates,
             }
 
-    def _get_job_ids(self, req: JobRequest = None):
+    def _get_job_ids(self, req: JobRequest = None) -> List[str]:
+        """
+        Extract the job IDs from a job request object
+
+        :param req: the job request to take the IDs from, defaults to None
+        :type req: JobRequest, optional
+
+        :return: list of job IDs
+        :rtype: List[str]
+        """
+        if not req.has_job_ids():
+            raise JobRequestException(NO_INPUT_TYPE_ERR)
         if req.has_batch_id():
             return self._jm.update_batch_job(req.batch_id)
-
-        try:
-            return req.job_id_list
-        except Exception as ex:
-            raise JobRequestException(ONE_INPUT_TYPE_ONLY_ERR) from ex
```

Comment:
Not sure what this was for? removed it

Reply:
If you look at the implementation of `req.job_id_list`, it returns `req.job_id` as a list if there was no `job_id_list` parameter.

Reply:
Yeah but `req.job_id_list` doesn't raise any exceptions except for its own `JobRequestException`?

Reply:
Then `JOBS_NOT_PROVIDED_ERR` should be `"job_id or job_id_list not provided"`?

Reply:
What if the request had a `batch_id` but no `job_id` or `job_id_list`? Then it did provide an input